packages/social-metadata-previews/src/editor/SocialPreviewEditor.js

```diff
@@ -0,0 +1,199 @@
+/* External dependencies */
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+/* Internal dependencies */
+import { SocialMetadataPreviewForm } from "@yoast/social-metadata-forms";
+import FacebookPreview from "../facebook/FacebookPreview";
+import TwitterPreview from "../twitter/TwitterPreview";
+import { recommendedReplacementVariablesShape, replacementVariablesShape } from "@yoast/replacement-variable-editor";
+
+/**
+ * A form with an image selection button, a title input field and a description field and the social preview.
+ *
+ * @returns {void} Void.
+ */
+class SocialPreviewEditor extends Component {
+	/**
+	 * The constructor.
+	 * @param {Object} props The props object.
+	 *
+	 * @returns {void}
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			activeField: "",
+			hoveredField: "",
+		};
+
+		this.SocialPreview = props.socialMediumName === "Facebook" ? FacebookPreview : TwitterPreview;
+		this.setHoveredField = this.setHoveredField.bind( this );
+		this.setActiveField = this.setActiveField.bind( this );
+		this.setEditorRef = this.setEditorRef.bind( this );
+		this.setEditorFocus = this.setEditorFocus.bind( this );
+	}
+
+	/**
+	 * Sets the field that the mouse is hovering over in state.
+	 *
+	 * @param {string} field The field that is hovered over.
+	 *
+	 * @returns {void}
+	 */
+	setHoveredField( field ) {
+		if ( field === this.state.hoveredField ) {
+			return;
+		}
+		this.setState( {
+			hoveredField: field,
+		} );
+	}
+
+	/**
+	 * Sets the active field that is selected in state.
+	 *
+	 * @param {string} field The field that is selected.
+	 *
+	 * @returns {void}
+	 */
+	setActiveField( field ) {
+		if ( field === this.state.activeField ) {
+			return;
+		}
+		this.setState(
+			{ activeField: field },
+			() => this.setEditorFocus( field )
+		);
+	}
+
+	/**
+	 * Sets focus on the editor that is the active field.
+	 *
+	 * @param {String} field The active field belonging to the editor to focus.
+	 *
+	 * @returns {void}
+	 */
+	setEditorFocus( field ) {
+		switch ( field ) {
+			case "title":
+				this.titleEditorRef.focus();
+				return;
+			case "description":
+				this.descriptionEditorRef.focus();
+				return;
+		}
+	}
+
+	/**
+	 * Sets the reference of each editor.
+	 * Used by child components to communicate with this focus managing component.
+	 * This component can then call the .focus() function on the passed refs.
+	 *
+	 * @param {String} field The field belonging to the editor that belongs to the ref.
```

Comment:
We are using all-lowercase names for our primitives (and that also seems to be the general convention).

```suggestion
	 * @param {string} field The field belonging to the editor that belongs to the ref.
```